relay-general/src/pii/regexes.rs

```diff
@@ -52,11 +53,13 @@ pub fn get_regex_for_rule_type(
     let kv = PatternType::KeyValue;
 
     match ty {
-        RuleType::RedactPair(ref redact_pair) => smallvec![(
-            kv,
-            &redact_pair.key_pattern.0,
-            ReplaceBehavior::replace_value()
-        )],
+        RuleType::RedactPair(ref redact_pair) => match redact_pair.key_pattern.compiled() {
+            Ok(pattern) => smallvec![(kv, pattern, ReplaceBehavior::replace_value())],
+            Err(err) => {
+                relay_log::error!("Unable to compile regex: {}", LogError(&err));
```

Comment:
Note that this is called for _every_ field in _every_ event we're processing. This can easily lead to a humongous number of error logs. It would be better to log this error just once at the compilation site.

Patterns can be based on user-input, so it's not clear to me whether we should even log this as error. It's expected that invalid patterns can be part of the project config.